File: .claude/skills/zero-trust-secret-orchestration/references/mjs/src/strategies/cloud.mjs
import { Strategy, SecretResolutionError, StrategyConfigError } from "../strategy.mjs";

export class CloudSecretStrategy extends Strategy {
  /**
   * Concrete subclasses implement `_fetch(key)`. Throwing from `_fetch`
   * propagates as a SecretResolutionError; returning `null` continues
   * the chain.
   */
  async _fetch(_key) {
    throw new Error("CloudSecretStrategy._fetch must be implemented by a subclass");
  }

  async get(key) {
    try {
      return await this._fetch(key);
    } catch (err) {
      if (err instanceof StrategyConfigError) throw err;
      throw new SecretResolutionError(`cloud fetch failed for ${key}`, { cause: err });
    }
  }
}

export class AwsSecretsManagerStrategy extends CloudSecretStrategy {
  constructor({ client } = {}) {
    super();
    this.client = client;
  }

  async _fetch(key) {
    if (!this.client) {
      throw new StrategyConfigError(
        "AwsSecretsManagerStrategy is a stub. Inject `{ client: smClient }` " +
          "from `@aws-sdk/client-secrets-manager` to activate it.",
      );
    }
    const out = await this.client.send({ SecretId: key });
    return out.SecretString ?? null;
  }
}

File: .claude/skills/zero-trust-secret-orchestration/references/py/src/zts/strategies/cloud.py
from zts.strategy import SecretResolutionError, Strategy, StrategyConfigError


class CloudSecretStrategy(Strategy):
    """Concrete subclasses implement `_fetch(key)`."""

    async def _fetch(self, key: str) -> str | None:
        raise NotImplementedError(
            "CloudSecretStrategy._fetch must be implemented by a subclass"
        )

    async def get(self, key: str) -> str | None:
        try:
            return await self._fetch(key)
        except StrategyConfigError:
            raise
        except Exception as exc:
            raise SecretResolutionError(f"cloud fetch failed for {key}") from exc


class AwsSecretsManagerStrategy(CloudSecretStrategy):
    def __init__(self, *, client=None):
        self.client = client

    async def _fetch(self, key: str) -> str | None:
        if self.client is None:
            raise StrategyConfigError(
                "AwsSecretsManagerStrategy is a stub. "
                "Inject `client=boto3.client('secretsmanager')` to activate it."
            )
        out = await self.client.get_secret_value(SecretId=key)
        return out.get("SecretString")
